app/util/notifications/hooks/index.test.ts

```diff
@@ -0,0 +1,249 @@
+import notifee, {
+  EventType,
+  Event as NotifeeEvent,
+} from '@notifee/react-native';
+
+import useNotificationHandler from './index';
+import { NavigationProp, ParamListBase } from '@react-navigation/native';
+import Routes from '../../../constants/navigation/Routes';
+import { Notification } from '../../../util/notifications/types';
+import { TRIGGER_TYPES } from '../constants';
+
+jest.mock('../../../util/device');
+jest.mock('../../../core/NotificationManager', () => ({
+  setTransactionToView: jest.fn(),
+}));
+jest.mock('@notifee/react-native', () => ({
+  setBadgeCount: jest.fn(),
+  decrementBadgeCount: jest.fn(),
+  onForegroundEvent: jest.fn(),
+  createChannel: jest.fn(),
+  EventType: {
+    DISMISSED: 'dismissed',
+    DELIVERED: 'delivered',
+    PRESS: 'press',
+  },
+  AndroidImportance: {
+    HIGH: 'high',
+  },
+}));
+
+jest.mock('../../../core/NotificationManager', () => ({
+  setTransactionToView: jest.fn(),
+}));
+
+jest.mock('../../../util/device', () => ({
+  isAndroid: jest.fn(),
+}));
+
+const mockNavigate = jest.fn();
+const mockNavigation = {
+  navigate: mockNavigate,
+} as unknown as NavigationProp<ParamListBase>;
+
+const notification = {
+  id: 1,
+  type: TRIGGER_TYPES.ERC1155_RECEIVED,
+  data: {
+    id: 1,
+    trigger_id: '1',
+    chain_id: 1,
+    block_number: 1,
+    block_timestamp: '',
+    tx_hash: '',
+    unread: false,
+    created_at: '',
+    address: '',
+    type: TRIGGER_TYPES.ERC1155_RECEIVED,
+    data: {},
+    createdAt: '',
+    isRead: false,
+  },
+} as unknown as Notification;
+
+const mockNotificationEvent = (event: NotifeeEvent) => ({
+  type: event.type,
+  detail: {
+    notification,
+  },
+});
+
+describe('useNotificationHandler', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should navigate to NOTIFICATIONS.DETAILS if notification is pressed', async () => {
+    const { result } = renderHook(() => useNotificationHandler(mockNavigation));
+
+    await result.current.handlePressedNotification(notification);
+
+    expect(mockNavigation.navigate).toHaveBeenCalledWith(
+      Routes.NOTIFICATIONS.DETAILS,
+      {
+        notificationId: notification.id,
+      },
+    );
+  });
+
+  it('should handle notifications correctly', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.PRESS,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  });
+
+  it('should do nothing if the EventType is DISMISSED', async () => {
```

Comment:
```suggestion
  it('does nothing if the EventType is DISMISSED', async () => {
```
Q: How do you define and test "doing nothing"?
The test `expect(notifee.onForegroundEvent).toHaveBeenCalled();` is the exact same as the previous test case where it displays. So either this one is not testing tht nothing happens, or both are not testing the right thing.

Reply:
I ended up refactoring this file so the only relevant tests were kept since some were indeed testing cases of NotificationsService, which has its own tests.